app/controllers/posts_controller.rb

```diff
@@ -16,4 +16,41 @@ def show
       flash.now[:error] = 'User not found'
     end
   end
+
+  def create
+    @user = current_user
+    @post = @user.posts.build(post_params)
+    @post.comments_counter = 0
+    @post.likes_counter = 0
+    if @post.save
+      flash[:success] = 'Post created successfully!'
+    else
+      flash[:error] = 'Post not created'
+    end
+
+    redirect_back(fallback_location: user_path(params[:user_id]))
+  end
+
+  def createcomment
+    @user = current_user
+    @post = @user.posts.find(params[:id])
+    @comment = @post.comments.build(comment_params)
+    @comment.author_id = @user.id
+    if @comment.save
+      flash[:success] = 'Comment created successfully!'
+    else
+      flash[:error] = 'Comment not created'
+    end
+    redirect_to user_post_path(@user.id, @post.id)
+  end
+
```

Comment:
- [ ] Kindly create `likes_controller` also to create `like`.
```
class LikesController < ApplicationController
    ...

    def create
      @post = Post.find(params[:post_id])
      @like = @current_user.likes.new(post_id: @post.id)

      respond_to do |format|
        if @like.save
          format.html { redirect_to @post }
        else
          format.html { redirect_to @post }
        end
      end
    end
	  ...

  end
```

Reply:
@Deepakdanger Hello thanks for your review but in project requirements there's no where we have to apply like functionality thanks you can check again